c2corg_ui/static/js/documentservice.js

```diff
@@ -57,6 +54,35 @@ app.Document = function(appAuthentication) {
   };
 };
 
+
+/**
+ * @param {appx.Document} doc
+ * @export
+ */
+app.Document.prototype.setDocument = function(doc) {
+  for (var attr in doc) {
+    if (attr === 'associations') {
+      this.setAssociations(doc.associations);
+    } else {
+      this.document[attr] = doc[attr];
+    }
+  }
+};
+
+
+/**
+ * @param {appx.DocumentAssociations} associations
+ * @export
+ */
+app.Document.prototype.setAssociations = function(associations) {
```

Comment:
It is used to fill the ``documentService``'s ``associations`` attribute from the module variable passed in the document view page. E.g. https://github.com/c2corg/v6_ui/blob/association-tool-fix-2/c2corg_ui/templates/waypoint/view.html#L66-L71